homeassistant/components/atag/climate.py

```diff
@@ -0,0 +1,132 @@
+    async_add_devices([AtagThermostat(coordinator, ENTITY_TYPES[CLIMATE])])
+
+
+class AtagThermostat(AtagEntity, ClimateDevice, RestoreEntity):
+    """Atag climate device."""
+
+    def __init__(self, coordinator, atagtype):
+        """Initialize with fake on/off state."""
+        self._on = None
+        super().__init__(coordinator, atagtype)
+
+    async def async_added_to_hass(self):
+        """Register callbacks & state restore for fake "Off" mode."""
+        await super().async_added_to_hass()
+        last_state = await self.async_get_last_state()
+        if last_state:
+            self._on = last_state.state != HVAC_MODE_OFF
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return SUPPORT_FLAGS
+
+    @property
+    def hvac_mode(self) -> Optional[str]:
+        """Return hvac operation ie. heat, cool mode."""
+        if not self._on:
+            return HVAC_MODE_OFF
+        return self.coordinator.atag.hvac_mode
+
+    @property
+    def hvac_modes(self) -> List[str]:
+        """Return the list of available hvac operation modes."""
+        return [HVAC_MODE_HEAT, HVAC_MODE_AUTO, HVAC_MODE_OFF]
+
+    @property
+    def hvac_action(self) -> Optional[str]:
+        """Return the current running hvac operation."""
+        if self.coordinator.atag.cv_status:
+            return CURRENT_HVAC_HEAT
+        return CURRENT_HVAC_IDLE
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        return TEMP_CELSIUS
+
+    @property
+    def current_temperature(self) -> Optional[float]:
+        """Return the current temperature."""
+        return self.coordinator.atag.temperature
+
+    @property
+    def target_temperature(self) -> Optional[float]:
+        """Return the temperature we try to reach."""
+        return self.coordinator.atag.target_temperature
+
+    @property
+    def max_temp(self):
+        """Return the maximum temperature."""
+        return DEFAULT_MAX_TEMP
+
+    @property
+    def min_temp(self):
+        """Return the minimum temperature."""
+        return DEFAULT_MIN_TEMP
+
+    @property
+    def preset_mode(self) -> Optional[str]:
+        """Return the current preset mode, e.g., auto, manual, fireplace, extend, etc."""
+        return self.coordinator.atag.hold_mode
+
+    @property
+    def preset_modes(self) -> Optional[List[str]]:
+        """Return a list of available preset modes."""
+        return SUPPORT_PRESET
+
+    async def async_set_temperature(self, **kwargs) -> None:
+        """Set new target temperature."""
+        if self._on and await self.coordinator.atag.set_temp(
+            kwargs.get(ATTR_TEMPERATURE)
+        ):
+            await self.coordinator.async_refresh()
+
+    async def async_set_hvac_mode(self, hvac_mode: str) -> None:
+        """Set new target hvac mode."""
+        self._on = hvac_mode != HVAC_MODE_OFF
+        if self._on:
```

Comment:
Is it not possible to turn the device off?

Reply:
It doesn't have that functionality, therefore using the "fake" approach to enable such a state in HA

Reply:
If it doesn't support that, why do we need to model it? Our standard is to only model the features that the device supports.

Reply:
Well I saw there is code for fake on/off in the climate integration. This makes it easier for a user to have a master switch to disable managing the device from HA (for example during summer, or to prevent erroneous sliding of the climate card in Lovelace on mobile, etc)
But I wasn't fully sure this is allowed, so if this needs to be removed say the word. A user would then have to enable an additional input Boolean to get the same result :)
Either way is fine with me!

Reply:
The "fake" turn_off service in the climate integration relies on the entity to actually support `HVAC_MODE_OFF`.

The mentioned use cases are just frontend concerns that we shouldn't take into consideration here in the backend.

Please remove `HVAC_MODE_OFF` if the device doesn't support it.

Reply:
Got it, will rework a bit tomorrow. Thanks for the review again!

Reply:
Done!